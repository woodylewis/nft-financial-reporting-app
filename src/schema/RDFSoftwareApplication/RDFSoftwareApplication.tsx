import React, { Fragment, FunctionComponent } from 'react';
import PropTypes from 'prop-types';

import { SoftwareApplication } from 'schema-dts';
import { JsonLd } from 'react-schemaorg';

interface RDFSoftwareApplicationProps {
  id: string;
  mainEntityId: string;
}

const RDFSoftwareApplication: FunctionComponent<RDFSoftwareApplicationProps> = ({
  id,
  mainEntityId,
}) => {
  const context = 'https://schema.org';
  const type = 'SoftwareApplication';

  return (
    <Fragment>
      <JsonLd<SoftwareApplication>
        item={{
          '@context': context,
          '@id': id,
          '@type': type,
          mainEntity: {
            '@id': mainEntityId,
            '@type': 'Consortium',
          },
        }}
      />
    </Fragment>
  );
};

RDFSoftwareApplication.propTypes = {
  id: PropTypes.string.isRequired,
  mainEntityId: PropTypes.string.isRequired,
};

export default RDFSoftwareApplication;
