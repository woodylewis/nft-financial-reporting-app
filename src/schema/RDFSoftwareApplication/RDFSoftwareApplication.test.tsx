import React from 'react';
// import { shallow, ShallowWrapper, mount } from "enzyme";
import { create, ReactTestRenderer } from 'react-test-renderer';

// import { JsonLd } from "react-schemaorg";
import RDFSoftwareApplication from './RDFSoftwareApplication';

// let wrapper: ShallowWrapper;
let snapshot: ReactTestRenderer;
// let mountedInstance;

beforeEach(() => {
  const app = <RDFSoftwareApplication id='DSFController' mainEntityId='SNNetwork' />;

  snapshot = create(app);
  // wrapper = shallow(app);
  // mountedInstance = mount(app);
});

describe('<RDFSoftwareApplication />', () => {
  test('it matches the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
