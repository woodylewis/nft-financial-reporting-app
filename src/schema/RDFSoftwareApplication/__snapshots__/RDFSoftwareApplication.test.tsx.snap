// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RDFSoftwareApplication /> it matches the snapshot 1`] = `
<script
  dangerouslySetInnerHTML={
    {
      "__html": "{"@context":"https://schema.org","@id":"DSFController","@type":"SoftwareApplication","mainEntity":{"@id":"SNNetwork","@type":"Consortium"}}",
    }
  }
  type="application/ld+json"
/>
`;
